File: 04.Simulation模拟/leetcode_3379_constructTransformedArray.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> constructTransformedArray(vector<int>& nums) {
        int n = nums.size();
        vector<int> res(n, 0);
        for (int i=0; i<n; i++){
            res[i] = nums[(((i+nums[i])%n)+n)%n];
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<int> nums = {3,-2,1,1};
    vector<int> res = s.constructTransformedArray(nums);
    for (int i : res){
        cout << i << " ";
    }
    cout << endl;
}

File: 04.Simulation模拟/leetcode_3379_constructTransformedArray.py
from typing import List

class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        res = [0]*n
        for i in range(n):
            res[i] = nums[(i+nums[i])%n]
        return res

if __name__ == "__main__":
    s = Solution()
    nums = [3,-2,1,1]
    print(s.constructTransformedArray(nums))
